cra-network-aware-service-worker-loading/src/sw-template.js

```diff
@@ -51,38 +51,41 @@ const CACHE_VERSION = 1;
 
 // Shorthand identifier mapped to specific versioned cache.
 const CURRENT_CACHES = {
-  DYNAMIC_NAME: 'ect-dynamic-v' + CACHE_VERSION
+  DYNAMIC_NAME: 'responsive-images-v' + CACHE_VERSION
 };
 
-const matchFunction = ({url, event}) => {
-  return new RegExp('max-res|medium-res|min-res|4g-video');
-};
+const expirationPlugin = new workbox.expiration.Plugin({
+  maxEntries: 50,
+  maxAgeSeconds: 24 * 60 * 60,
+});
 
-const expirationManager = new workbox.expiration.CacheExpiration(
-  CURRENT_CACHES.DYNAMIC_NAME,
-  {
-    maxAgeSeconds: 24 * 60 * 60, // 1 Day
-    maxEntries: 50
-  }
-);
+// See https://developers.google.com/web/tools/workbox/guides/handle-third-party-requests#force_caching_of_opaque_responses
+const cacheOpaqueResponsesPlugin = new workbox.cacheableResponse.Plugin({
+  statuses: [0, 200]
+});
 
-const handler = async ({url, event}) => {
-  console.log('[sw handler] requesting event.request.url => ', event.request.url);
-  const cache = await caches.open(CURRENT_CACHES.DYNAMIC_NAME);
-  const response = await cache.match(event.request);
-  if (response) {
-    console.log('[sw handler] returning match-cached response');
-    return response;
-  } else {
-    console.log('[sw handler] returning fetched response');
-    const networkResponse = await fetch(event.request);
-    cache.put(event.request, networkResponse.clone());
-    await expirationManager.updateTimestamp(event.request.url);
-    return networkResponse;
+// See https://developers.google.com/web/tools/workbox/guides/using-plugins#custom_plugins
+const normalizeCacheKeyPlugin = {
+  cacheKeyWillBeUsed: async ({request, mode}) => {
+    // Normalize the cache key.
+    // So 'https://example.com/image-max-res.jpg' and 'https://example.com/image-min-res.jpg'
+    // would both end up using the same normlized cache key, 'ect-responsive-images'
+    // 'https://example.com/4g-video.mp4', 'ect-responsive-video'
+    // Because we ignore `mode`, this will be done for both reads and writes.
+    const normalizedCacheKey = request.url.search(/max-res|medium-res|min-res/) > -1 ? 'ect-responsive-images' : 'ect-responsive-video';
```

Comment:
This will work for a demo in which there's exactly one media file used, but for a real-world app, with multiple media files that each might have `*-res` as part of their URLs, they'll end up overwriting each other since they'll all share the same cache key.

My original suggestion is to normalize the cache key by removing the part that changes based on quality (`max-res`, `medium-res`, etc.) while keeping the rest of the URL, which will presumably be unique for each semantically different image.

Reply:
Thank you for your feedback @jeffposnick.

https://cdn.glitch.com/8d7fb7f0-a9be-4a8c-96c7-8af286af487e%2Fmin-res.jpg?v=1562842586912
https://cdn.glitch.com/8d7fb7f0-a9be-4a8c-96c7-8af286af487e%2Fmedium-res.jpg?v=1562842587169
https://cdn.glitch.com/8d7fb7f0-a9be-4a8c-96c7-8af286af487e%2Fmax-res.jpg?v=1562842587982
https://cdn.glitch.com/8d7fb7f0-a9be-4a8c-96c7-8af286af487e%2F4g-video.mp4?v=1562842601068

Here if we remove the part that changes based on quality ( `max-res`, `medium-res`, etc.) there is id part like `v=` so I thought it was not normalized because the URL stays different for each ECT even if we remove that quality part and results in trying to read only its own resource not normalized one.

I got your points. I'll change the URLs so that we can normalize. e.g removing id part.
I'll let you know once I've got it done.

Reply:
Hi @jeffposnick.

I've tried to implement your suggestion correctly.
Could you please review the implementation?
Thanks